pkg/storage/stores/shipper/indexshipper/tsdb/index/chunk.go

```diff
@@ -25,6 +26,39 @@ func (c ChunkMeta) From() model.Time                 { return model.Time(c.MinTi
 func (c ChunkMeta) Through() model.Time              { return model.Time(c.MaxTime) }
 func (c ChunkMeta) Bounds() (model.Time, model.Time) { return c.From(), c.Through() }
 
+// Sub subtracts the other chunk from c if the overlap. If the other is completely
+func Sub(l, r *ChunkMeta) []*ChunkMeta {
```

Comment:
note, this is unused and even probably incorrect.

I think the assumption is l.Min > r.Min and l.Max < r.Max, but then on line 37 we should do r.Max-l.Max right?